test/e2e/storage/detach_mounted.go

```diff
@@ -0,0 +1,205 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package storage
+
+import (
+	"fmt"
+	"math/rand"
+	"path"
+
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/storage/utils"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var (
+	BusyBoxImage = imageutils.GetE2EImage(imageutils.BusyBox)
+)
+
+var _ = utils.SIGDescribe("Detaching volumes", func() {
+	f := framework.NewDefaultFramework("flexvolume")
+
+	// note that namespace deletion is handled by delete-namespace flag
+
+	var cs clientset.Interface
+	var ns *v1.Namespace
+	var node v1.Node
+	var suffix string
+
+	BeforeEach(func() {
+		framework.SkipUnlessProviderIs("gce", "local")
+		framework.SkipUnlessMasterOSDistroIs("debian", "ubuntu", "gci", "custom")
+		framework.SkipUnlessNodeOSDistroIs("debian", "ubuntu", "gci", "custom")
+		framework.SkipUnlessSSHKeyPresent()
+
+		cs = f.ClientSet
+		ns = f.Namespace
+		nodes := framework.GetReadySchedulableNodesOrDie(f.ClientSet)
+		node = nodes.Items[rand.Intn(len(nodes.Items))]
+		suffix = ns.Name
+	})
+
+	It("should not work when mount is in progress", func() {
+		driver := "attachable-with-long-format"
+		driverInstallAs := driver + "-" + suffix
+
+		By(fmt.Sprintf("installing flexvolume %s on node %s as %s", path.Join(driverDir, driver), node.Name, driverInstallAs))
+		installFlex(cs, &node, "k8s", driverInstallAs, path.Join(driverDir, driver))
+		By(fmt.Sprintf("installing flexvolume %s on master as %s", path.Join(driverDir, driver), driverInstallAs))
+		installFlex(cs, nil, "k8s", driverInstallAs, path.Join(driverDir, driver))
+		volumeSource := v1.VolumeSource{
+			FlexVolume: &v1.FlexVolumeSource{
+				Driver: "k8s/" + driverInstallAs,
+			},
+		}
+
+		clientPod := getFlexVolumePod(volumeSource, node.Name)
+		By("Creating pod that uses slow format volume")
+		pod, err := cs.CoreV1().Pods(ns.Name).Create(clientPod)
+		Expect(err).NotTo(HaveOccurred())
+
+		uniqueVolumeName := getUniqueVolumeName(pod, driverInstallAs)
+
+		By("waiting for volume-in-use on the node after pod creation")
+		err = waitForVolumesInUse(cs, node.Name, uniqueVolumeName)
+		Expect(err).NotTo(HaveOccurred(), "while waiting for volume in use")
+
+		By("Force deleting the flexvolume pod")
+		err = cs.CoreV1().Pods(ns.Name).Delete(pod.Name, metav1.NewDeleteOptions(0))
```

Comment:
do you need force delete? Graceful delete should have the same behavior?

Reply:
right but I am just trying to cause the detach operation to start sooner.

Reply:
hmm, I do not think that will a difference to test case either ways.

Reply:
I just to make sure because sometimes controller reacts differently when using force delete instead of graceful delete.

Reply:
alright I have removed the force delete option from here.